### PR TITLE
Add maven publishing

### PR DESCRIPTION
**Pull request by wmontwe**
## Description

Adds 3 new deploys, which are pushed to https://github.com/d4l-data4life/maven-repository:

- release -> whenever a version is tagged
- snapshot -> always latest state of main will be available
- feature -> for every feature branch there will be a version using the branch name as identifier -> `1.2.0-add-maven-publishing-SNAPSHOT`

Added issue template that was missing in this project

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by me-d4l — DISMISSED** [review]
Thanks for introducing this 🎊 I have not evaluated the changes in detail, but looking good in general.

File: .github/workflows/d4l-ci-publish-release.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   push:
     tags:
       - 'v*'
 
 jobs:
-  build:
+  publish-release:
 
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, macos]
```
Comment (bitPogo) [inline .github/workflows/d4l-ci-publish-release.yml:11]: Comparing with the SDK one, I am not quite understand, why this is self hosted and the other not - could explain me the reason for this, please?

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -18,6 +18,8 @@ toc::[]
 
 === Changed
 
+* BREAKING module name changed from `hc-fhir-sdk-java` to `fhir-java`
```
Comment (bitPogo) [inline CHANGELOG.adoc:21]: Is that related to this patch? I currently struggling to figure out, where this is happening.

**Review by bitPogo — COMMENTED** [review]
Looks good over all, but 2 minor things.

File: CHANGELOG.adoc (hunk 2 above)
Comment (wmontwe) [reply CHANGELOG.adoc:21]: the old publishing config had an error using the project name instead of the module name, which resulted in a dependency name `care.data4life.hc-fhir-sdk-java:hc-fhir-sdk-java` and I fixed it in the new config to also use the same pattern as in all other SDKs

File: .github/workflows/d4l-ci-publish-release.yml (hunk 1 above)
Comment (wmontwe) [reply .github/workflows/d4l-ci-publish-release.yml:11]: Then we need to patch the SDK one too. The release is patching another GitHub Repository and this is only possible with a SSH key of the user that is patching. For the self-hosted machines it is already configured. The public runners need to be configured in that way but that requires some extra effort.

Comment (bitPogo) [reply .github/workflows/d4l-ci-publish-release.yml:11]: Thx!

File: CHANGELOG.adoc (hunk 2 above)
Comment (bitPogo) [reply CHANGELOG.adoc:21]: 👍

**Review by bitPogo — APPROVED** [review]
Looks good to me
